### PR TITLE
[OS Patch] PyTorch 1.11 

### PR DESCRIPTION
**Pull request by junpuf**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/buildspec.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
 region: &REGION <set-$REGION-in-environment>
 framework: &FRAMEWORK pytorch
-version: &VERSION 1.12.1
-short_version: &SHORT_VERSION "1.12"
+version: &VERSION 1.11.0
+short_version: &SHORT_VERSION "1.11"
```
Comment (saimidu) [inline pytorch/buildspec.yml:5]: Please ensure that the `pytorch/buildspec-1-11.yml` file is usable to build these images, before this PR is merged, so that we can ensure the config in the buildspec file used by the  pipeline is correct.

**Comment by saimidu** [comment]
Tests seem to be failing.

File: dlc_developer_config.toml [hunk 2]
```diff
@@ -23,32 +23,32 @@ benchmark_mode = false
...
 ### On by default
 sanity_tests = true
   safety_check_test = false
   ecr_scan_allowlist_feature = false
-ecs_tests = true
-eks_tests = true
-ec2_tests = true
+ecs_tests = false
+eks_tests = false
+ec2_tests = false
```
Comment (saimidu) [inline dlc_developer_config.toml:40]: Why disable these tests?

Comment (junpuf) [reply dlc_developer_config.toml:40]: was just copying the example, if can turn those on.

File: pytorch/buildspec.yml (hunk 1 above)
Comment (junpuf) [reply pytorch/buildspec.yml]: updated

**Comment by saimidu** [comment]
Change looks good. Waiting to see test results.

**Comment by junpuf** [comment]
#2268

**Comment by junpuf** [comment]
#2270

File: test/dlc_tests/eks/pytorch/training/test_eks_pytorch_training.py [hunk 3]
```diff
@@ -58,14 +58,14 @@ def test_eks_pytorch_single_node_training(pytorch_training):
...
-      mv $FILE examples/mnist/main.py
+      sed -i '1d' mnist/main.py &&
+      sed -i '6d' mnist/main.py &&
+      cat mnist/main.py >> $FILE &&
+      rm mnist/main.py &&
+      mv $FILE mnist/main.py
     '''
 
-    args = f"git clone https://github.com/pytorch/examples.git && {mnist_dataset_download_config}  && python examples/mnist/main.py"
+    args = f"git clone https://github.com/pytorch/examples.git && cd examples && git reset --hard 5a06e9cac1728c860b53ebfc6792e0a0e21a5678 && {mnist_dataset_download_config}  && python mnist/main.py"
```
Comment (saimidu) [inline test/dlc_tests/eks/pytorch/training/test_eks_pytorch_training.py:68]: Why `git reset --hard` and not `git checkout`?

Comment (junpuf) [reply test/dlc_tests/eks/pytorch/training/test_eks_pytorch_training.py:68]: new commit broke the test

Comment (kace) [inline test/dlc_tests/eks/pytorch/training/test_eks_pytorch_training.py:68]: We probably have to come up w/ a better solution for our examples, but pinning to a specific commit works for now.

**Review by kace — APPROVED** [review]
Need to address merge conflicts, but LGTM otherwise.

**Review by kace — APPROVED** [review]
Still good after resolving merge conflicts!

**Review by kace — APPROVED** [review]
Approved.
